zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/compensation/CompensationEventExecutionTest.java

```diff
@@ -2493,6 +2493,157 @@ public void shouldNotTriggerCompensationForChildProcess() {
         .doesNotContain(tuple(CHILD_PROCESS_ID, "Undo-B", ProcessInstanceIntent.ELEMENT_ACTIVATED));
   }
 
+  @Test
+  public void shouldInvokeMultiInstanceActivityCompensationHandler() {
+    // given
+    final Consumer<BoundaryEventBuilder> compensationHandler =
+        compensation ->
+            compensation
+                .serviceTask("Undo-A")
+                .zeebeJobType("Undo-A")
+                .multiInstance()
+                .zeebeInputCollectionExpression("[1,2,3]");
+
+    final BpmnModelInstance process =
+        Bpmn.createExecutableProcess(PROCESS_ID)
+            .startEvent()
+            .serviceTask(
+                "A",
+                task -> task.zeebeJobType("A").boundaryEvent().compensation(compensationHandler))
+            .endEvent()
+            .compensateEventDefinition()
+            .done();
+
+    ENGINE.deployment().withXmlResource(process).deploy();
+
+    final long processInstanceKey = ENGINE.processInstance().ofBpmnProcessId(PROCESS_ID).create();
+
+    // when
+    ENGINE.job().ofInstance(processInstanceKey).withType("A").complete();
+
+    completeJobs(processInstanceKey, "Undo-A", 3);
+
+    // then
+    assertThat(
+            RecordingExporter.processInstanceRecords()
+                .withProcessInstanceKey(processInstanceKey)
+                .limitToProcessInstanceCompleted())
+        .extracting(
+            r -> r.getValue().getBpmnElementType(),
+            r -> r.getValue().getBpmnEventType(),
```

Comment:
💭 Might be better to add also the `elementId`, to better underline that the compensation handler is completed 3 times